opte/src/engine/port.rs

```diff
@@ -1410,7 +1415,7 @@ impl<N: NetworkImpl> Port<N> {
         data: &mut PortData,
         pmeta: &PacketMeta,
         pkt_len: u64,
-    ) -> result::Result<TcpState, String> {
+    ) -> result::Result<TcpState, ProcessError> {
```

Comment:
I would use a separate error here so you don't need all the `unreachable!` cases for errors that don't apply. You can then add that as a variant to `ProcessError` if you want.